File: csharp/Validators/ArgumentValueValidator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Validators
{
    public class ArgumentValueValidator: AbstractValidator
    {
        protected new static readonly string ValidationFailureMessage = "NOTE: the value of argument {0} should be {1}!";
        private ushort Index { get; set; }
        private string Value { get; set; }

        public ArgumentValueValidator(List<string> args, ushort index, string value) : base(args)
        {
            Index = index;
            Value = value;
        }

        public override bool Validate()
        {
            if (Arguments[Index] != Value)
            {
                Console.WriteLine(string.Format(ValidationFailureMessage, Index, Value));
                return false;
            }

            return true;
        }
    }
}

File: java/JavaFileutil/src/JavaFileutil/Validators/ArgumentValueValidator.java
package JavaFileutil.Validators;

import java.util.List;

public class ArgumentValueValidator extends AbstractValidator
{
    private short Index;
    private String Value;

    public ArgumentValueValidator(List<String> args, short index, String value, String message)
    {
        super(args, message);
        Index = index;
        Value = value;
    }

    @Override
    public boolean Validate()
    {
        if(Arguments.size() <= Index || !Arguments.get(Index).equals(Value))
        {
            System.out.println(ErrorPreface + Message);
            return false;
        }

        return true;
    }
}
